lib/active_merchant/billing/gateways/litle.rb

```diff
@@ -62,19 +72,24 @@ def credit(money, authorization, options = {})
         refund(money, authorization, options)
       end
 
-      def refund(money, authorization, options={})
-        transaction_id, _, _ = split_authorization(authorization)
-
+      def refund(money, payment, options={})
         request = build_xml_request do |doc|
           add_authentication(doc)
           add_descriptor(doc, options)
-          doc.credit(transaction_attributes(options)) do
-            doc.litleTxnId(transaction_id)
-            doc.amount(money) if money
+          doc.send(refund_type(payment), transaction_attributes(options)) do
+            if payment.is_a?(String) || payment.is_a?(Numeric)
+              transaction_id, kind, _ = split_authorization(payment)
+              doc.litleTxnId(transaction_id)
+              doc.amount(money) if money
+            elsif check?(payment)
+              add_echeck_purchase_params(doc, money, payment, options)
+            else
+              add_auth_purchase_params(doc, money, payment, options)
+            end
```

Comment:
Maybe it's the naming of the methods being used in here in `refund` (`add_echeck_purchase_params` and `add_auth_purchase_params`) but do refunds require the exact same params as echeck and auth purchases?

Reply:
@dtykocki Litle allows for "credits" of non-vantiv (litle) transactions. In those cases you give it the same info as a purchase and they "credit" the amount instead.